### ISSUE TITLE
Docker compose mounted volumes interfere with dockerization / packing

### ISSUE THREAD

**Issue by jaimergp** [Quansight/conda-store#73]
I am running `main` on a MBP M1. Running `docker compose up --build` with the current configuration (`data/` exposed in different `volumes`) results in two kinds of errors.

__Conda pack__

Default configuration (with volumes) will error out with:

```
conda-store-build_1   | Traceback (most recent call last):
conda-store-build_1   |   File "/opt/conda-store-server/conda_store_server/build.py", line 209, in conda_build
conda-store-build_1   |     build_conda_pack(conda_store, build_path, build)
conda-store-build_1   |   File "/opt/conda-store-server/conda_store_server/build.py", line 241, in build_conda_pack
conda-store-build_1   |     conda.conda_pack(prefix=conda_prefix, output=output_filename)
conda-store-build_1   |   File "/opt/conda-store-server/conda_store_server/conda.py", line 25, in conda_pack
conda-store-build_1   |     conda_pack.pack(prefix=str(prefix), output=str(output))
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_pack/core.py", line 514, in pack
conda-store-build_1   |     env = CondaEnv.from_prefix(prefix,
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_pack/core.py", line 163, in from_prefix
conda-store-build_1   |     files = load_environment(prefix, **kwargs)
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_pack/core.py", line 856, in load_environment
conda-store-build_1   |     raise CondaPackException(_missing_files_error.format(packages))
conda-store-build_1   | conda_pack.core.CondaPackException: 
conda-store-build_1   | Files managed by conda were found to have been deleted/overwritten in the
conda-store-build_1   | following packages:
conda-store-build_1   | 
conda-store-build_1   | - ncurses 6.2:
conda-store-build_1   |     share/terminfo/2/2621A
conda-store-build_1   |     share/terminfo/E/Eterm
conda-store-build_1   |     share/terminfo/E/Eterm-color
conda-store-build_1   |     + 1034 others
```

Commenting out the volumes in `docker-compose.yaml` fixes it. Adding `ignore_missing_files=True` to the `conda_pack.pack` call fixes it too, but probably in an unwanted way.

__Conda docker__

This one will fail with or without volumes, apparently due to a missing temporary folder that got deleted too early?

```
conda-store-build_1   | [2021-06-29 07:19:33,490] {/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/conda.py:635} INFO - building conda environment
[2021-06-29 07:19:41,118] {/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/utils.py:12} INFO - building conda environment took 7.627 [s]
conda-store-build_1   | [2021-06-29 07:19:41,118] {/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/conda.py:536} INFO - adding conda environment in package layers
conda-store-build_1   | [2021-06-29 07:19:41,393] {/opt/conda-store-server/conda_store_server/build.py:214} ERROR - [Errno 2] No such file or directory: '/tmp/tmpktw3xrfe/opt/conda/conda-meta/_libgcc_mutex-0.1-conda_forge.json'
conda-store-build_1   | Traceback (most recent call last):
conda-store-build_1   |   File "/opt/conda-store-server/conda_store_server/build.py", line 210, in conda_build
conda-store-build_1   |     build_docker_image(conda_store, build_path, build)
conda-store-build_1   |   File "/opt/conda-store-server/conda_store_server/build.py", line 263, in build_docker_image
conda-store-build_1   |     image = build_docker_environment_image(
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/conda.py", line 646, in build_docker_environment_image
conda-store-build_1   |     add_conda_layers(
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/conda.py", line 581, in add_conda_layers
conda-store-build_1   |     add_conda_package_layers(
conda-store-build_1   |   File "/opt/conda/envs/conda-store-server/lib/python3.9/site-packages/conda_docker/conda.py", line 548, in add_conda_package_layers
conda-store-build_1   |     with open(meta_path) as f:
conda-store-build_1   | FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpktw3xrfe/opt/conda/conda-meta/_libgcc_mutex-0.1-conda_forge.json'
conda-store-build_1   | [2021-06-29 07:19:41,424] {/opt/conda-store-server/conda_store_server/build.py:57} INFO - specification name=dfgfgdfd build has failed=2 times
```

The only way around this is to disable the Docker image creation by commenting that line out.

Not sure how well this can be reproduced in other OS / archs.

**Comment by jaimergp** [comment]
I think the `conda-pack` error has to do with APFS (Apple's Filesystem) being case-insensitive by default.

**Comment by jaimergp** [comment]
Did some more digging and I don't think it's worth an upstream patch. It's a very weird corner case of virtualizing Docker on top of a case-insensitive filesystems _and_ adding a mounted volume. I added an environment variable in #79 so people can skip the missing files check on a case by case basis.

**Comment by jaimergp** [comment]
With respect to the `conda-docker` error, I think the issue might lie within [these lines](https://github.com/conda-incubator/conda-docker/blob/master/conda_docker/conda.py#L468-L486). Changing that to `subprocess.run` and manually raising the exception when `retcode != 0` results in an error as expected. I am trying to replace `conda-standalone` with `micromamba` to see if that's the reason. 

My assumption is that the files are never unpacked in the expected directory. They are indeed downloaded, but extraction must be failing (or target location is different) because I tried browsing the temporary folder and didn't see any of the expected files there, hence the "file not found" errors.

Unfortunately I ran out of time today, so I'll get back to it next week!
